flask_mongorest/resources.py

```diff
@@ -83,7 +96,7 @@ def raw_data(self):
                     raise ValidationError({'error': "Chunked Transfer-Encoding is not supported."})
 
                 try:
-                    self._raw_data = json.loads(request.data)
+                    self._raw_data = json.loads(request.data, parse_constant=self._enforce_strict_json)
```

Comment:
That's more of a side effect than the core issue. We simply shouldn't allow invalid values to ever be accepted and stored. That being said, I agree it makes sense to also add `allow_nan` because without it, an exception isn't raised during serialization, and we're not notified about the invalid data.